Extends plugin-helpers to be used for automating version changes (#2398)
opensearch-project/OpenSearch-Dashboards: packages/osd-plugin-helpers/README.md

## Usage

> This CLI has a `build` command that plugin devs can run to easily package OpenSearch Dashboards plugins. Called with the `version`
> command, a plugin's manifest and package files can be updated to match the version of OpenSearch Dashboards or one supplied.

Reviewer: I think we can make it more clear that there are two commands and what they do:
```suggestion
This CLI has a `build` command that plugin devs can run to easily package OpenSearch Dashboards plugins. It also has a `version`
command, which updates a plugin's version in manifest and package files to match the version of OpenSearch Dashboards or one supplied.
```

  Usage: plugin-helpers [command] [options]

  Commands:
    build
      Copies files from the source into a zip archive that can be distributed for
      installation into production OpenSearch Dashboards installs. The archive includes the non-
      development npm dependencies and builds itself using raw files in the source
      directory so make sure they are clean/up to date. The resulting archive can
      be found at:

        build/{plugin.id}-{opensearchDashboardsVersion}.zip

      Options:
        --skip-archive                       Don't create the zip file, just create the build/opensearch-dashboards directory

>         --opensearch-dashboards-version, -v  OpenSearch Dashboards version that the

Reviewer 2: is it incomplete help text?

Author: It is the output of the CLI `—help`

Reviewer: Yeah, this was the existing help text, but we should go ahead and update it to complete the sentence - something like:

```suggestion
        --opensearch-dashboards-version, -v  OpenSearch Dashboards version that the built plugin will target
```

>         --opensearch-dashboards-version, -v  OpenSearch Dashboards version that the built plugin will target

Author: What a terrible decision. Sadly changing this would be a "breaking change" and will need to be scheduled for a major release.

     
    version

>       Updates the version to a provided parameter, or syncs it with the version of OpenSearch Dashboards

Reviewer: It may be clear to users who are familiar with this tool, but I think we could be more explicit about which files are updated and where. Is it something I'd generally run before `build`, after running `build` (to update the version in just the artifacts), or is it more of a tool for development?

Reviewer 2: assume this will change plugin package.json file which is also source code. a follow up CR is required to commit the change. If so, maybe provide a friendly remind to develop who used the tool.

  
      Options:
        --sync        Update the versions to match Dashboards' (default)

>         --set         Update the version to a specific value

Reviewer: Can we provide an example that shows the version formatting expected?

Author: Updated the string to call out that a "semantic version" is expected

>         --sync        Update the versions to match Dashboards' (default)

Author: Originally, I did that intentionally. I have reverted that in the latest iteration.

>         --set         Update the version to a specific semantic version value

Reviewer 3: I agree with josh, a simple example here will definitely be more helpful. semantic is well understood but but given our x.x.x.x versioning system it may be a little confusing as to what we expect as the semantic version. Also its a lot easier with examples

Author: I added some text.
